agr_literature_service/global_utils.py

```diff
@@ -40,3 +48,56 @@ def wrapper(*args, **kwargs):
             return f(*args, **kwargs)
     wrapper.has_run = False
     return wrapper
+
+
+def get_next_curie(subdomain, db=None):  # pragma: no cover
+
+    if environ.get('ENV_STATE') and environ.get('ENV_STATE') != 'test':
+        token = get_authentication_token()
+        headers = generate_headers(token)
+        headers['subdomain'] = subdomain
+        url = environ['ID_MATI_URL']
+        headers['value'] = '1'
+        res = requests.post(url, headers=headers)
+        res_json = res.json()
+        new_curie = res_json['first']['curie']
+        return new_curie
+    return get_next_local_curie(subdomain, db)
+
+
+def get_next_local_curie(subdomain, db):
+
+    ### it is only for testing purpose
+    # db_session = db
+    # if db is None:
+    #    db_session = create_postgres_session(False)
+    curie_start = "AGRKB:102"
+    rs = None
+    if subdomain == 'reference':
+        curie_start = "AGRKB:101"
+        rs = db.execute("SELECT curie FROM reference order by reference_id desc")
```

Comment:
@sweng66 is it any more efficient to add a LIMIT 1 to only get one ?

Reply:
Good idea! Initially I thought this is just for unit tests so it only has a few references in the test db anyway, but I think it is a good practice to do so! Thanks for the suggestion! I will fix it @azurebrd